client/lib/features-list/utils.js

```diff
@@ -0,0 +1,14 @@
+function featureNotPartOfTrial( feature ) {
+	return feature.not_part_of_free_trial;
+}
+
+function featureListHasAFreeTrialException( featuresList ) {
+	return featuresList.some( function( feature ) {
```

Comment:
You don't need to create a new anonymous function here:

```js
return featuresList.some( featureNotPartOfTrial );
```

Also, this function is only used once, and `featuresList.some( featureNotPartOfTrial );` by itself is easy to read and understand. What do you think of getting rid of this function altogether?